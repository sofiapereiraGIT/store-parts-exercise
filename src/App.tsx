import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect,
} from "react-router-dom";
import Header from "./components/common/Header";
import ItemsList from "./pages/ItemsList";
import ItemDetails from "./pages/ItemDetails";
import { App as S } from "./styles";

function App() {
  return (
    <Router>
      <Header />
      <S.PageContentWrapper>
        <Switch>
          <Route exact path="/" component={ItemsList} />
          <Route exact path="/item" component={ItemDetails} />
          <Redirect to={"/"} />
        </Switch>
      </S.PageContentWrapper>
    </Router>
  );
}

export default App;
