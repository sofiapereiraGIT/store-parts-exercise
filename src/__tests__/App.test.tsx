import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

describe("App Tests", () => {
  it("Renders properly", () => {
    const renderedComponent = renderer.create(<App />);
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });
});
