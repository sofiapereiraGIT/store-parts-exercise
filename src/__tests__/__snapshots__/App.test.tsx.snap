// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests Renders properly 1`] = `
Array [
  <header
    className="sc-bczRLJ cMfLWl"
  >
    <h1
      className="sc-gsnTZi bVxfwA"
      data-testid="headerTitle"
      onClick={[Function]}
    >
      Store Parts
    </h1>
  </header>,
  <div
    className="sc-kgflAQ bocrLy"
  >
    <div
      className="sc-kDDrLX kAIxVA"
    >
      <h2>
        Items List
      </h2>
      <div
        className="sc-dkzDqf blxtaX"
      >
        <div
          className="MuiFormControl-root MuiTextField-root sc-hKMtZM lmpSen"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="search"
            id="search-label"
          >
            Search
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch"
              data-testid="searchField"
              disabled={false}
              id="search"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="search"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Search
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          aria-owns={null}
          className="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon sc-eCYdqJ PVeSc css-16awh2u-MuiAutocomplete-root"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
        >
          <div
            className="MuiFormControl-root MuiTextField-root sc-hKMtZM lmpSen MuiFormControl-fullWidth"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="autocomplete"
              id="autocomplete-label"
            >
              Type
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-activedescendant={null}
                aria-autocomplete="list"
                aria-expanded={false}
                aria-invalid={false}
                autoCapitalize="none"
                autoComplete="off"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                disabled={false}
                id="autocomplete"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                placeholder=""
                readOnly={false}
                required={false}
                role="combobox"
                spellCheck="false"
                type="text"
                value=""
              />
              <div
                className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Type
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium sc-jSMfEi cJyXvP css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="descOrderIcon"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="arrowDownwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m20 12-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </button>
      </div>
      <table
        aria-label="customized table"
        className="MuiTable-root sc-iBkjds eHvXh css-rqglhn-MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root sc-ftvSup fRrNtf css-15wwp11-MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium sc-papXJ hcCvrO css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium sc-papXJ hcCvrO css-1ygcj2i-MuiTableCell-root"
              scope="col"
            >
              Type
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium sc-papXJ hcCvrO css-1azl6jz-MuiTableCell-root"
              scope="col"
            >
              Price
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          role={null}
        />
      </table>
      <div
        className="sc-gKXOVf eDBIUH"
      >
        <span
          data-testid="noDataMessage"
        >
          No data available
        </span>
      </div>
    </div>
  </div>,
]
`;
